Gems/Terrain/Code/Source/TerrainRenderer/TerrainFeatureProcessor.cpp

```diff
@@ -480,27 +480,40 @@ namespace Terrain
         m_detailMaterialManager.SetDetailMaterialConfiguration(config);
     }
     
-    void TerrainFeatureProcessor::SetWorldSize([[maybe_unused]] AZ::Vector2 sizeInMeters)
+    void TerrainFeatureProcessor::SetMeshConfiguration(const MeshConfiguration& config)
     {
-        // This will control the max rendering size. Actual terrain size can be much
-        // larger but this will limit how much is rendered.
+        m_meshManager.SetConfiguration(config);
     }
     
-    void TerrainFeatureProcessor::CacheForwardPass()
+    void TerrainFeatureProcessor::CachePasses()
     {
+        m_passes.clear();
         auto rasterPassFilter = AZ::RPI::PassFilter::CreateWithPassClass<AZ::RPI::RasterPass>();
         rasterPassFilter.SetOwnerScene(GetParentScene());
         AZ::RHI::RHISystemInterface* rhiSystem = AZ::RHI::RHISystemInterface::Get();
-        AZ::RHI::DrawListTag forwardTag = rhiSystem->GetDrawListTagRegistry()->AcquireTag(AZ::Name("forward"));
+        const AZ::RHI::DrawListTag forwardTag = rhiSystem->GetDrawListTagRegistry()->AcquireTag(AZ::Name("forward"));
+        const AZ::RHI::DrawListTag depthTag = rhiSystem->GetDrawListTagRegistry()->AcquireTag(AZ::Name("depth"));
+        const AZ::RHI::DrawListTag shadowTag = rhiSystem->GetDrawListTagRegistry()->AcquireTag(AZ::Name("shadow"));
+
         AZ::RPI::PassSystemInterface::Get()->ForEachPass(rasterPassFilter,
             [&](AZ::RPI::Pass* pass) -> AZ::RPI::PassFilterExecutionFlow
             {
                 auto* rasterPass = azrtti_cast<AZ::RPI::RasterPass*>(pass);
 
-                if (rasterPass && rasterPass->GetDrawListTag() == forwardTag)
+                if (rasterPass)
                 {
-                    m_forwardPass = rasterPass;
-                    return AZ::RPI::PassFilterExecutionFlow::StopVisitingPasses;
+                    if (rasterPass->GetDrawListTag() == forwardTag)
+                    {
+                        m_passes.push_back(rasterPass);
+                    }
+                    if (rasterPass->GetDrawListTag() == depthTag)
+                    {
+                        m_passes.push_back(rasterPass);
+                    }
+                    if (rasterPass->GetDrawListTag() == shadowTag)
+                    {
+                        m_passes.push_back(rasterPass);
+                    }
```

Comment:
Nit: These should be else-ifs, they can't all apply to the same pass, right?

Reply:
Tempted to make this a switch statement since the tags are just handles which are just uints, but using the index directly feels a bit nasty.
```
switch (rasterPass->GetDrawListTag().GetIndex())
{
    ...
}
```
I'll just do if-else